code/modules/projectiles/guns/energy.dm

```diff
@@ -754,3 +754,63 @@
 	fire_delay = 0.15 SECONDS
 	rounds_per_shot = 3
 	message_to_user = "You set the machine laser's charge mode to efficiency mode."
+
+/obj/item/weapon/gun/energy/lasgun/lasrifle/xray
+	name = "\improper Terra Experimental X-Ray laser rifle"
+	desc = "A Terra Experimental X-Ray laser rifle, abbreviated as the TE-X. It has an integrated charge selector for normal and high settings. Uses standard Terra Experimental (abbreviated as TE) power cells. As with all TE Laser weapons, they use a lightweight alloy combined without the need for bullets any longer decreases their weight and aiming speed quite some vs their ballistic counterparts."
+	reload_sound = 'sound/weapons/guns/interact/standard_laser_rifle_reload.ogg'
+	fire_sound = 'sound/weapons/guns/fire/laser3.ogg'
+	icon_state = "tex"
+	item_state = "tex"
+	max_shots = 40 //codex stuff
+	ammo_datum_type = /datum/ammo/energy/lasgun/marine/xray
+	rounds_per_shot = 15
+	attachable_allowed = list(
+		/obj/item/attachable/bayonet,
+		/obj/item/attachable/bayonetknife,
+		/obj/item/attachable/reddot,
+		/obj/item/attachable/lasersight,
+		/obj/item/attachable/flashlight,
+		/obj/item/attachable/magnetic_harness,
+		/obj/item/attachable/scope/marine,
+		/obj/item/attachable/scope/mini,
+		/obj/item/weapon/gun/flamer/mini_flamer,
+		/obj/item/attachable/motiondetector,
+		/obj/item/attachable/buildasentry,
```

Comment:
itd be cool if you made the buildasentry ignore walls if its on xray mode. If you wanna figure out how to do it, look at the code for the sentry ignoring windows with energy bullets.

Reply:
Wouldnt it be kinda, too good? I thought BAS already shoots anything within LOS even if its behind a wall

Reply:
the corner issue is cuz of differences of how bullets travel vs how the sentry determines whats in range. However this might give the gun a reason to attach BAS

Reply:
I might add it later in different PR if i feel like it